### PR TITLE
Add modalities filter option to readPolicies

### PR DESCRIPTION
**Pull request by BigIdeas**
## 📚 Purpose
FHV conductor requires the ability to filter policies by `modalities`. This adds support for that to `mds-policy-service`. A future PR will bring this new filter param into `mds-gql`.

Any policy with at least one rule matching any of the requested modalities will be returned.

While working on this I noticed that in city-dev there are policies without rules, policies whose rules are an empty array, and policies with rules that have no `modality`. I'm not sure whether we should expect policies like those in production, but I wrote the `modality` query such that those will all be returned by a query for `micromobility` (the default modality) just in case.

## 🗄️ Database Changes:
- N/A

## 💔 Breaking Changes:
- N/A

### CODE REVIEW

**Comment by janedotx** [comment]
@BigIdeas We're considering having policies with no rules be deactivating policies, and if we go through with that, they shouldn't ever show up in the front-end.

File: packages/mds-policy-service/repository/index.ts [hunk 1]
```diff
@@ -145,6 +146,19 @@ export const PolicyRepository = ReadWriteRepository.Create('policies', { entitie
...
       }
 
+      if (modalities) {
+        query.andWhere(
+          "EXISTS(SELECT FROM json_array_elements(policy_json->'rules') elem WHERE elem->>'modality' = ANY(:modalities))",
+          { modalities }
+        )
+        if (modalities.includes('micromobility')) {
+          // Policies without rules or with any rule whose modality == null are micromobility by default
+          query.orWhere(
```
Comment (janedotx) [inline packages/mds-policy-service/repository/index.ts:156]: I think this is all right. The modality is assumed to be `micromobility` if not explicitly stated in a rule.

**Review by janedotx — APPROVED** [review]

**Review by cjlynch12 — APPROVED** [review]

**Review by avatarneil — COMMENTED** [review]
Does this need a corresponding AJV query param validation update too?

**Comment by BigIdeas** [comment]
> Does this need a corresponding AJV query param validation update too?

Good question but... I don't think so? I'm not seeing any existing validation of the readPolicies query params. Very happy to write a quick validator if we're moving in that direction though.

**Review by nessur — APPROVED** [review]
Nice!

**Comment by nessur** [comment]
> > Does this need a corresponding AJV query param validation update too?
> 
> Good question but... I don't think so? I'm not seeing any existing validation of the readPolicies query params. Very happy to write a quick validator if we're moving in that direction though.
@avatarneil yeah, checked, none of the queries on Policy Service validate their input. 
In `mds-policy-api` there's a bunch of parsers, but they're express-style parsers, not AJV-ish things.

**Comment by BigIdeas** [comment]
> @BigIdeas We're considering having policies with no rules be deactivating policies, and if we go through with that, they shouldn't ever show up in the front-end.

Awesome, good to know! I'm thinking I'll keep this PR as is now - err on the side of showing all policies - and can easily delete the extra query for null rules if/when we start using those to deactivate policies.

**Pull request merged**